x-pack/plugins/fleet/server/services/files/types.ts

```diff
@@ -10,24 +10,9 @@ import type { Readable } from 'stream';
 import type { BaseFileMetadata, FileCompression, FileStatus } from '@kbn/shared-ux-file-types';
 
 /**
- * The type of file.
- * Use `from-host` when interacting with files that were sent to ES from the
- * host (via Fleet-Server)
- * Use `to-host` when interacting with files that are being sent to the host
- * (via fleet-server)
+ * Interface for files that were created by a host and consumed in kibana
  */
-export type FleetFileTransferDirection = 'from-host' | 'to-host';
-
-export interface FleetFileClientInterface {
-  /** Creates a new file. Only applicable when type of file is `to-host`. */
-  create(fileStream: HapiReadableStream, agentIds: string[]): Promise<FleetFile>;
-
-  /** Deletes a file. Only applicable when type of file is `to-host`. */
-  delete(fileId: string): Promise<void>;
-
-  /** Updates metadata for the file. Only applicable when type of file is `to-host`. */
-  update(fileId: string, updates: Partial<FleetFileUpdatableFields>): Promise<FleetFile>;
-
+export interface FleetFromHostFileClientInterface {
```

Comment:
the two interfaces could be unnecessary I think - there is no polymorphism, no multiple implementations, we just have two separate classes. what do you think?

Reply:
What do you mean?
The two different implementations have different interfaces and I did not want to expose the Class as a type, but rather, create a Type that the Class `implements`.

Reply:
why don't you want to expose the class? it's practically a type I guess, and it's already exported/exposed, so can be used as the return type for `createFilesClient.toHost()` and `...fromHost()` functions

I mean, as `interface1 === class1` and `interface2 === class2`, I don't see a reason for this abstraction and hence the increased complexity. if there is another class to implement the same interface, then the abstraction makes sense.

Reply:
The class is not exported. It is private to Fleet and should only be instantiated by Fleet. Exporting it would allow others to consume it outside of the control of Fleet.

Re: types -vs- class

I have come across the two different opinions: that one should not just create a class with no interface as well as every class should have a well defined interface. Honestly, I tend to use a combination of the two, but I do lean more to having well defined types that classes then `implements`. 

if you are ok with it, I will leave it as is.

Reply:
oh! I missed that the class should be only used inside Fleet, and isn't exported via the `fleet/server/index.ts`. the who-can-import-what guidelines are a bit blurry to me, but I see your point - other plugins won't be able to instantiate an interface.

thanks a lot for the explanation, I agree leaving it this way.